qiskit/pulse/model/frames.py

```diff
@@ -0,0 +1,197 @@
+    """
+
+    def __init__(self, identifier: str, frequency: float, phase: Optional[float] = 0.0):
+        """Create ``GenericFrame``.
+
+        Args:
+            identifier: A unique identifier used to identify the frame.
+            frequency: The initial frequency set for the frame.
+            phase: The initial phase set for the frame. Default value 0.
+
+        """
+        self._frequency = frequency
+        self._phase = phase
+        super().__init__(identifier)
+
+    @property
+    def name(self) -> str:
+        """Return the name of the frame."""
+        return self._identifier
+
+    @property
+    def frequency(self) -> float:
+        """Return the initial frequency of the generic frame."""
+        return self._frequency
+
+    @property
+    def phase(self) -> float:
+        """Return the initial phase of the generic frame."""
+        return self._phase
+
+    def __eq__(self, other: "GenericFrame") -> bool:
+        """Return True iff self and other are equal, specifically, iff they have the same type
+        name, frequency and phase.
+
+        Args:
+            other: The generic frame to compare to this one.
+
+        Returns:
+            True iff equal.
+        """
+        return (
+            type(self) is type(other)
+            and self._identifier == other._identifier
+            and self._frequency == other._frequency
+            and self._phase == other._phase
+        )
+
+    def __hash__(self) -> int:
+        return self._hash
+
+
+class QubitFrame(Frame):
+    """A frame associated with the driving of a qubit.
+
+    :class:`QubitFrame` is a frame associated with the driving of a specific qubit.
+    The initial frequency of
+    the frame will be taken as the default driving frequency provided by the backend
+    during compilation.
+    """
+
+    def __init__(self, qubit_index: int):
+        """Create ``QubitFrame``.
+
+        Args:
+            qubit_index: The index of the qubit represented by the frame.
+        """
+        self._validate_index(qubit_index)
+        super().__init__(qubit_index)
+
+    @property
+    def qubit_index(self) -> int:
+        """Return the qubit index of the qubit frame."""
+        return self._identifier
+
+    def _validate_index(self, index) -> None:
+        """Raise a ``PulseError`` if the qubit index is invalid. Namely, check if the index is a
+        non-negative integer.
+
+        Raises:
+            PulseError: If ``identifier`` (index) is a negative integer.
+        """
+        pass
+        if not isinstance(index, (int, np.integer)) or index < 0:
+            raise PulseError("Qubit index must be a non-negative integer")
+
+
+class MeasurementFrame(Frame):
+    """A frame associated with the dispersive readout of a qubit.
+
+    ``MeasurementFrame`` is a frame associated with the dispersive readout of a specific qubit,
+    which requires a stimulus tone driven at frequency off resonant to qubit drive.
```

Comment:
We can be more general than dispersive readout here.

```suggestion
    """A frame associated with the measurement of a qubit.

    ``MeasurementFrame`` is a frame associated with the readout of a specific qubit,
    which requires a stimulus tone driven at frequency off resonant to qubit drive.
```